File: python-demo/ollama/ollama_cloud.py
import os
from ollama import Client

# 获取 API Key 并检查
api_key = os.environ.get('OLLAMA_API_KEY')
if not api_key:
    raise ValueError("请设置 OLLAMA_API_KEY 环境变量")

client = Client(
    host="https://ollama.com",
    headers={'Authorization': 'Bearer ' + api_key}
)

messages = [
  {
    'role': 'user',
    'content': '天空为啥是蓝色的?',
  },
]

for part in client.chat('gpt-oss:120b', messages=messages, stream=True):
  print(part['message']['content'], end='', flush=True)

File: js-demo/ollama/ollama_cloud.js
require('dotenv').config();
const ollama = require('ollama');

const client = new ollama.Ollama({
  host: 'https://ollama.com',
  headers: {
    'Authorization': 'Bearer ' + process.env.OLLAMA_API_KEY
  }
});

async function chatStream() {
  const messages = [
    {
      role: 'user',
      content: '天空为啥是蓝色的?'
    }
  ];

  try {
    const stream = await client.chat({
      model: 'gpt-oss:120b',
      messages: messages,
      stream: true
    });

    for await (const part of stream) {
      process.stdout.write(part.message.content);
    }
    console.log();
  } catch (error) {
    console.error('Error:', error);
  }
}

chatStream();
